Source/WebCore/html/HTMLDialogElement.cpp

```diff
@@ -177,4 +198,21 @@ void HTMLDialogElement::setIsModal(bool newValue)
     m_isModal = newValue;
 }
 
+void HTMLDialogElement::queueDialogToggleEventTask(DialogState oldState, DialogState newState)
```

Comment:
This should be shared somehow with the details / popover code to avoid duplication.

Reply:
I had considered this but I also thought it might be better as a follow up PR. Thoughts?

Reply:
I would prefer refactoring details / popover first and then adopting the new shared code here.

Reply:
https://github.com/WebKit/WebKit/pull/23361